test/test_runner.rb

```diff
@@ -0,0 +1,37 @@
+require 'test/unit'
+require 'mocha/test_unit'
+require_relative '../lib/pmdtester/builders/diff_builder'
+require_relative '../lib/pmdtester/builders/html_report_builder'
+require_relative '../lib/pmdtester/builders/pmd_report_builder'
+require_relative '../lib/pmdtester/runner'
+
+# Unit test class for PmdTester::Runner
+class TestRunner < Test::Unit::TestCase
+  include PmdTester
+  def test_local_mode
+    report_diff = ReportDiff.new
+    PmdReportBuilder.any_instance.stubs(:build).returns(nil).twice
```

Comment:
I need to get used to this style :smile: The solution in java would look completely different (you would inject mocks into the Runner module) - but in scripting languages, you can redefine everything....